### ISSUE TITLE
New Rule: Headings with only whitespace (WCAG 2.0 A)

### ISSUE THREAD

**Issue by jeeyyy** [dequelabs/axe-core#1419]
For Rule Specification see 
- https://github.com/auto-wcag/auto-wcag/pull/400

## Rule:
- heading-whitespace-only

## Check (any):
- heading-whitespace-only

**Comment by jkodu** [comment]
# AT Testing of examples from above spec

## Pass Scenario Examples:

| Scenario/ Platform | Firefox (NVDA) | IE (Jaws) | Safari (VoiceOver) | Chrome (Talkback) |
|---|---|---|---|---|
| `<h2></h2>` | Nothing is identified | Nothing is identified | Nothing is identified | Nothing is identified |
| `<div role="heading"></div>` | Nothing is identified | Nothing is identified | Nothing is identified | Nothing is identified |
| `<h2>'</h2>` | Identified as `heading level 2` | Identified as `heading level 2` | Identified as `apostrophe heading level 2` | Identified as `heading level 2` |

## Fail Scenario Examples:

| Scenario/ Platform | Firefox (NVDA) | IE (Jaws) | Safari (VoiceOver) | Chrome (Talkback) |
|---|---|---|---|---|
| `<h2><br /></h2>` | Nothing is identified | Nothing is identified | Identified as `2 heading level 2` | Nothing is identified |
| `<h2>&nbsp;</h2>` | Nothing is identified | Identified as `heading level 2` | Nothing is identified | Nothing is identified |
| `<h2> </h2>` | Nothing is identified | Nothing is identified | Nothing is identified | Nothing is identified |
| `<h2>&#32;</h2>` | Nothing is identified | Nothing is identified | Nothing is identified | Nothing is identified |
| `<h2>&ensp;</h2>` | Identified as `heading level 2` | Nothing is identified | Nothing is identified | Nothing is identified |
| `<h2>&emsp;</h2>` |Identified as `heading level 2` | Nothing is identified | Nothing is identified | Nothing is identified |
| `<h2>&thinsp;</h2>` |Identified as `heading level 2` | Nothing is identified | Nothing is identified | Nothing is identified |
| `<h2><span><br /></span></h2>` | Nothing is identified | Nothing is identified | Nothing is identified | Nothing is identified |

## Inapplicable Examples:

| Scenario/ Platform | Firefox (NVDA) | IE (Jaws) | Safari (VoiceOver) | Chrome (Talkback) |
|---|---|---|---|---|
| `<div></div>` | Nothing is identified | Nothing is identified | Nothing is identified | Nothing is identified |
| `<h2 aria-hidden="true"></h2>` | Nothing is identified | Nothing is identified | Nothing is identified | Nothing is identified |

**Comment by WilcoFiers** [comment]
Scenarios 3, 4 and 8 don't look like they should fail. If they get consistently ignored, axe-core shouldn't flag them.

**Comment by WilcoFiers** [comment]
Axe-core already has a heading-empty rule.
